feat: 2022-10-07 storage release notes (#238)
neondatabase/website: content/release-notes/2022-10-07.md

---
label: 'Storage'
---

### What's new

> - Pageserver: Increase default compaction_period setting to 20 s

Reviewer: ```suggestion
- Pageserver: Increased the default compaction_period setting to 20 seconds, affecting the per-tenant compaction background process. With numerous projects, the previous default setting of 1 second caused excessive CPU usage.
```

Reviewer 2: besides the increased CPU usage on pageserver that is not necessarily affecting the users, were there symptoms that users could feel that may be relieved with this?

Reviewer: Looks like the issue was found in captest. 
https://github.com/neondatabase/neon/issues/2542
https://github.com/neondatabase/neon/pull/2527
I have revised the entry with more details.

> - Pageserver: Initial support for online tenant rebalancing

Reviewer: ```suggestion
- Pageserver: Added initial support for online tenant rebalancing.
```

Reviewer: changed to relocation.

> - Pageserver: Support for multiple postgres versions (14 and 15 for now)

Reviewer: ```suggestion
- Pageserver: Added support for multiple PostgreSQL versions.
```

Reviewer: Do we want to announce support for PG 15 here?

Author: +1, let's keep it the way you've phrased it

> - Compute: Add `h3_pg` extension

Reviewer: ```suggestion
- Compute: Added support for the `h3_pg` extension, which provides PostgreSQL bindings for the [H3 Core Library](https://github.com/uber/h3).
```

> - Compute: Beta version of sequential scan page prefetch

Reviewer: ```suggestion
- Compute: Introduced a beta version of sequential scan page prefetch, which improves sequential scan performance.
```

Reviewer: Is the above addition accurate?

Author: It is, thanks!

Reviewer 2: experimental? early? not sure about beta.
also, is the prefetch enabled in production? this entry looks like it is. is it actually isn't smth like 'added support for experimental sequential scan prefetch implementation' may be more appropriate.

Reviewer 2: i am also thinking about the the readers who aren't familiar with db perf tuning
are there any links we can add here to help them relate to sequential scan performance?

Reviewer: How about: 
"Added support for a future implementation of sequential scan prefetch, which improves I/O performance for operations such as table scans."

Reviewer: Revised as mentioned above.

> - Compute: Backpressure algorithm tuning

Reviewer: Tuned the backpressure algorithm, which manages the lag between the Pageserver and compute node or the Pageserver and the Write-ahead Log (WAL) service.

> - Compute: Backpressure algorithm tuning

Reviewer: ```suggestion
- Compute: Tuned the backpressure algorithm, which manages the lag between the Pageserver and compute node or the Pageserver and the Write-ahead Log (WAL) service.
```

Reviewer 2: should we add 'backpressure' to glossary?
what are the symptoms that this particular adjustment improves or relieves?

Reviewer: Created a PR for "backpressure" glossary addition.
Updated the release note entry, which was reviewed by @knizhnik.

> - Proxy: Improve error messages and logging

Reviewer: ```suggestion
- Proxy: Improved error messages and logging.
```
